plugins/calibre.koplugin/parser.lua

```diff
@@ -1,90 +1,93 @@
--- A parser for metadata.calibre
-local util = require("util")
+-- parse "metadata.calibre" files
+local lj = require("lunajson")
 
--- removes leading and closing characters and converts hex-unicodes
-local function replaceHexChars(s, n, j)
-    local l = string.len(s)
-    if string.sub(s, l, l) == "\"" then
-        s = string.sub(s, n, string.len(s)-1)
-    else
-        s = string.sub(s, n, string.len(s)-j)
-    end
-    s = string.gsub(s, "\\u([a-f0-9][a-f0-9][a-f0-9][a-f0-9])", function(w)
-        return util.unicodeCodepointToUtf8(tonumber(w, 16))
-    end)
-    return s
-end
+local array_fields = {
+    authors = true,
+    tags = true,
+    series = true,
+}
 
--- a couple of string helper functions for dealing with raw json strings
-local function isEqual(str, key)
-    if str:sub(1, key:len() + 6) == string.format("    \"%s\"", key) then
-        return true
-    end
-    return false
-end
+local required_fields = {
+    authors = true,
+    last_modified = true,
+    lpath = true,
+    series = true,
+    series_index = true,
+    size = true,
+    tags = true,
+    title = true,
+    uuid = true,
+}
 
-local function getValue(str, key)
-    if str == string.format("    \"%s\": null, ", key) then
-        return nil
+local field
+local t = {}
+local function append(v)
+    -- These *may* be arrays, so we need the extra check to confirm that startarray ran
+    if array_fields[field] and t[field] then
```

Comment:
I realized we can simplify that some more ;).

```suggestion
    -- Some fields *may* be arrays, so check whether we ran through startarray first or not
    if t[field] then
```